packages/devtools_app/lib/src/memory/flutter/memory_heatmap.dart

```diff
@@ -10,37 +10,160 @@ import 'package:flutter/material.dart' hide TextStyle;
 import 'package:flutter/rendering.dart' hide TextStyle;
 import 'package:flutter/widgets.dart' hide TextStyle;
 import 'package:intl/intl.dart';
+import 'package:provider/provider.dart';
 
+import '../../flutter/auto_dispose_mixin.dart';
+
+import 'memory_controller.dart';
 import 'memory_graph_model.dart';
 
-class FlameChart extends StatelessWidget {
+class HeatMapSizeAnalyzer extends SingleChildRenderObjectWidget {
+  const HeatMapSizeAnalyzer({
+    Key key,
+    Widget child,
+  }) : super(key: key, child: child);
+
+  @override
+  RenderFlameChart createRenderObject(BuildContext context) {
+    return RenderFlameChart();
+  }
+}
+
+class RenderFlameChart extends RenderProxyBox {
+  @override
+  void paint(PaintingContext context, Offset offset) {
+    if (child != null) {
+      context.paintChild(child, offset);
+    }
+  }
+}
+
+class FlameChart extends StatefulWidget {
   const FlameChart(
-    this.sizes, {
-    // Flame chart has a blueish color.
-    this.lightColor = const Color(0xFFBBDEFB),
-    this.darkColor = const Color(0xFF0D47A1),
-  });
+    this.controller,
+  );
 
-  final InstructionsSize sizes;
-  final Color lightColor;
-  final Color darkColor;
+  final MemoryController controller;
+
+  @override
+  FlameChartState createState() => FlameChartState(controller);
+}
+
+class FlameChartState extends State<FlameChart> with AutoDisposeMixin {
+  FlameChartState(this.controller);
+
+  InstructionsSize sizes;
+
+  Map<String, Function> callbacks = {};
+
+  /// Flame chart has a blueish color.
+  Color lightColor = const Color(0xFFBBDEFB);
+
+  Color darkColor = const Color(0xFF0D47A1);
+
+  MemoryController controller;
+
+  Widget snapshotDisplay;
+
+  _FlameChart _flameChart;
+
+  @override
+  void initState() {
+    super.initState();
+  }
+
+  @override
+  void didChangeDependencies() {
+    super.didChangeDependencies();
+
+    controller = Provider.of<MemoryController>(context);
+
```

Comment:
short circuit return if `controller == Provider.of<MemoryController>(context)`

Reply:
Added a TODO this doesn't work only in the heatmap.